src/library/unfold_macros.cpp

```diff
@@ -119,8 +119,6 @@ class unfold_untrusted_macros_fn {
 };
 
 static bool contains_untrusted_macro(unsigned trust_lvl, expr const & e) {
-    if (trust_lvl > LEAN_BELIEVER_TRUST_LEVEL)
-        return false;
     return static_cast<bool>(find(e, [&](expr const & e, unsigned) {
```

Comment:
> we currently pass the untrusted macros to the kernel without unfolding them, and the kernel throws an error.

If this is the case, then removing this line is just masking the real problem. 
The kernel object should have been created using the same `trust_lvl` we are checking here. 
So, if the kernel is rejecting the macro, there is a bug somewhere. 
The first possibility that comes to my mind is: a macro was defined with `trust_lvl` greater than  `LEAN_BELIEVER_TRUST_LEVEL`.